realm/index.d.ts

```diff
@@ -107,24 +107,24 @@ declare namespace Realm {
          * @param  {any[]} ...arg
          * @returns Results
          */
-        filtered(query: string, ...arg: any[]): Results;
+        filtered(query: string, ...arg: any[]): Results<T & Object>;
```

Comment:
`T & Object` doesn't actually do anything. The apparent type of `T` is `{}` which is effectively the same as `Object` - in short, you get all the same members of `Object` anyway. Furthermore, `number` and `string` are assignable to `Object` and `{}`, so it doesn't do a ton for you.

I think what you really want to do is constrain `T` to be of only an object type, which TypeScript 2.2 will have. [See here](https://github.com/Microsoft/TypeScript/issues/1809)

Until then, I think you'll just have to roll with `T`.